modules/bindbackend/bindbackend2.cc

```diff
@@ -302,7 +302,8 @@ void Bind2Backend::getUpdatedMasters(vector<DomainInfo> *changedDomains)
     ReadLock rl(&s_state_lock);
 
     for(state_t::const_iterator i = s_state.begin(); i != s_state.end() ; ++i) {
-      if(!i->d_masters.empty() && this->alsoNotify.empty() && i->d_also_notify.empty())
+      if(i->d_kind != DomainInfo::Slave ||
+          (!i->d_masters.empty() && this->alsoNotify.empty() && i->d_also_notify.empty()))
```

Comment:
should these be removed or are Master domains with masters considered slaves?

Reply:
Good question, I'd have to check what the expected behaviour is.

Reply:
In the gsqlite3 backend, we `SELECT ... where type='MASTER'`. This also means that alsoNotify and d_masters don't even matter. We should not use NATIVE or SLAVE zones here to make the BIND backend behave consistent with the gsql backends. I'll make that so.